test/test_testing.py

```diff
@@ -4301,77 +4331,93 @@ def test_list_directory_object_itself(self):
+            client.make_dir(f"{pebble_dir}/subdir/subdir", make_parents=False)
         assert excinfo.value.kind == 'not-found'
 
-        client.make_dir(f"{self.prefix}/subdir/subdir", make_parents=True)
-        assert client.list_files(f"{self.prefix}/subdir", pattern='subdir')[0].path == \
-            f"{self.prefix}/subdir/subdir"
+        client.make_dir(f"{pebble_dir}/subdir/subdir", make_parents=True)
+        assert client.list_files(f"{pebble_dir}/subdir", pattern='subdir')[0].path == \
+            f"{pebble_dir}/subdir/subdir"
 
-    def test_make_directory_with_relative_path_fails(self):
-        client = self.client
+    def test_make_directory_with_relative_path_fails(
+        self,
+        client: typing.Union[_TestingPebbleClient, pebble.Client],
+    ):
         with pytest.raises(pebble.PathError) as excinfo:
             client.make_dir('dir')
         assert excinfo.value.kind == 'generic-file-error'
 
-    def test_make_subdir_of_file_fails(self):
-        client = self.client
-        client.push(f"{self.prefix}/file", 'data')
+    def test_make_subdir_of_file_fails(
+        self,
+        pebble_dir: str,
+        client: typing.Union[_TestingPebbleClient, pebble.Client],
+    ):
+        client.push(f"{pebble_dir}/file", 'data')
 
         # Direct child case
         with pytest.raises(pebble.PathError) as excinfo:
-            client.make_dir(f"{self.prefix}/file/subdir")
+            client.make_dir(f"{pebble_dir}/file/subdir")
         assert excinfo.value.kind == 'generic-file-error'
 
         # Recursive creation case, in case its flow is different
         with pytest.raises(pebble.PathError) as excinfo:
-            client.make_dir(f"{self.prefix}/file/subdir/subdir", make_parents=True)
+            client.make_dir(f"{pebble_dir}/file/subdir/subdir", make_parents=True)
         assert excinfo.value.kind == 'generic-file-error'
 
-    def test_make_dir_with_permission_mask(self):
-        client = self.client
-        client.make_dir(f"{self.prefix}/dir1", permissions=0o700)
-        client.make_dir(f"{self.prefix}/dir2", permissions=0o777)
+    def test_make_dir_with_permission_mask(
+        self,
+        pebble_dir: str,
+        client: typing.Union[_TestingPebbleClient, pebble.Client],
+    ):
+        client.make_dir(f"{pebble_dir}/dir1", permissions=0o700)
+        client.make_dir(f"{pebble_dir}/dir2", permissions=0o777)
 
-        files = client.list_files(f"{self.prefix}/", pattern='dir*')
-        assert [f for f in files if f.path == f"{self.prefix}/dir1"][0].permissions == 0o700
-        assert [f for f in files if f.path == f"{self.prefix}/dir2"][0].permissions == 0o777
+        files = client.list_files(f"{pebble_dir}/", pattern='dir*')
+        assert [f for f in files if f.path == f"{pebble_dir}/dir1"][0].permissions == 0o700
+        assert [f for f in files if f.path == f"{pebble_dir}/dir2"][0].permissions == 0o777
```

Comment:
In the original logic, `self.prefix` is a str, so I kept it as it is. Not sure which is better, because it seems there are other test cases that could benefit from it being a string. I will keep it as it is for now since refactoring a bunch of tests are out of the scope :)